src/main/java/org/apache/commons/math4/stat/descriptive/moment/Mean.java

```diff
@@ -211,7 +215,7 @@ public double evaluate(final double[] values, final int begin, final int length)
      */
     @Override
     public double evaluate(final double[] values, final double[] weights,
-                           final int begin, final int length) throws MathIllegalArgumentException {
+                           final @IndexFor({"#1", "#2"}) int begin, final @NonNegative @LTLengthOf(value = {"#1", "#2"}, offset = {"#3 - 1", "#3 - 1"}) int length) throws MathIllegalArgumentException {
```

Comment:
This method also indicates that the first two arguments must have the same length in the documentation.